### DISCUSSION TITLE
How do you handle exceptional cases while keeping your metadata up to date?

### DISCUSSION THREAD

**Discussion by afontenot** [beetbox/beets#4907] [General]
Anyone who has used MusicBrainz extensively for metadata tagging in Beets has surely encountered problem cases. A few examples, using my collection:

 * MB sets the artist name for some albums by the band [AJJ](https://musicbrainz.org/artist/5e1dbdb2-87ac-41ff-9960-4a42a872327a) as "Andrew Jackson Jihad", which is the former name of the band. I want all my albums by this band to be tagged "AJJ", since it's the current name of the band and I'd like all the albums to be categorized together. MB's tagging isn't *incorrect* since it represents the actual release name, but it won't work for me.
 * I don't like the join phrase `; ` used on some classical albums for the album artist, I prefer to manually edit it to something more descriptive.
 * [Some releases](https://musicbrainz.org/release/8d8e51b5-7d45-43bb-a045-d26cab2b950d) are worldwide with blackout regions. Some enterprising MusicBrainz user has added a release event for every country except those in which the release is unavailable, rather than using the standard "XW" release. Picard handles this by tagging a "US" event. Beets can't handle the situation and tags the file with the first country, "AF" (Afghanistan). I'd prefer to force it to say "XW" (worldwide) in these cases or leave it blank entirely.

Obviously, each person is going to have their own issues like this. They don't reflect necessarily reflect issues with the MB database - at least, we can't all go in and edit it to our personal tastes. This means maintaining a bunch of differences in your local files. Some will make the changes they want and never update their files, but I prefer to get the latest data from MusicBrainz every so often. I've learned interesting things about my music collection from this data.

As far as I can tell Beets offers no easy way to do this, and I'm wondering what solutions / workarounds others may have employed here.

I have a few ideas but I'm not really happy with them:

 * Set a custom label on albums that shouldn't be kept in sync. You can then do a re-import of your library with something like `beet import -L ^mbsync:skip`.
   * This has a bunch of downsides - you can't use mbsync, due to its limitations, and you lose the ability to get new metadata for any album that's been customized. 
 * Every time you `beet mod` something, append the command to a script. Then, after you mbsync, you can run the script again to restore your local changes
   * This seems messy, unreliable, and essentially means storing the important metadata outside of Beets.

From a development point of view, if I was going to improve this, it seems like I'd want to focus on the mbsync plugin. Does that seem reasonable? Might not be too hard to (a) make it possible to only update files that match a query, (b) prompt when any tag from a configured list would be updated rather than updating silently.

**Comment by arsaboo** [comment]
For the first issue, you can use the [rewrite](https://beets.readthedocs.io/en/latest/plugins/rewrite.html) plugin to rewrite the artist name. 

```yaml
rewrite:
    artist AJJ: Andrew Jackson Jihad
```
For the second issue, use the `importreplace` plugin:

```yaml
importreplace:
  replacements:
    - album_fields: artist
      replace:
        ' ; ': 'REPLACEMENT'
```

**Reply by afontenot** [reply]
Interesting ideas, thanks! The rewrite option might work for me. A potential downside is that it sounds like the plugin [is not very consistent](https://github.com/beetbox/beets/issues/2786#issuecomment-356677369) about whether it modifies the metadata - because it wasn't originally intended to.

The importreplace won't work, because I don't want a different join phrase, I need full customization. You know how MusicBrainz standards for popular music are to stylize artist credits in the artist tag ("Andrew Bird feat. Fiona Apple")? I take the same approach with classical music. So you need a way to use different joining phrases between different artists in a single tag. Obviously for classical music "feat." is not usually appropriate, but it captures the idea - the artist field is not just a joined version of the list of artists.

More generally, I'm just curious if there are any beets users that maintain tracks tagged differently than MusicBrainz wants them **and** intermittently update their tags against the MusicBrainz database.
